### DISCUSSION TITLE
Question about symbology and expressions

### DISCUSSION THREAD

**Discussion by sylvain258** [maplibre/maplibre-gl-js#3871] [Q&A]
Hi,

I am working on datae and I am trying to create a symbology using expressions. Problem is at 'circle-color', I have indicated that white is a default color, but in my app, the rendering is black.

I don't understand why, this is my code.

Thank you very much for your help,

Sylvain

`const date0 = ["<=", ['get', 'annee'], 1904]; `
          `const date1 = ['all', ['>', ['get', 'annee'], 1904], ['<=', ['get', 'annee'], 1945]]; `
          `const date2 = ['all', ['>', ['get', 'annee'], 1945], ['<=', ['get', 'annee'], 1975]];`
          `const date3 = [">=", ['get', 'annee'], 1975];`
          `const daten = ["==", ['get', 'annee'], null]; // possible that the problem is here (with coalesce ?)`
        
     `     map.addLayer({
            'id': 'CPAAPI',
            "type": 'circle',
            'source': 'cpa',
            'source-layer': 'cpa_carte',
            'layout': {},
            'maxzoom': 24,
            'minzoom': 1,
            "paint":{
              "circle-radius": [
                'interpolate',
                ['linear'],
                ['zoom'],
                5,1,
                7, 3,
                14, 5
              ],
              'circle-color': [
                'case',
                date0,
                "#cdcdcd", 
                date1,
                '#C58300', // marron
                date2,
                '#C500B6', // violet
                date3,
                '#B0C500',
                daten,
                '#e31a1c',
                '#fff' // app returns black
              ],
              "circle-stroke-width": 1,
              "circle-stroke-color": "#00353D"
            }
          });`

**Comment by HarelM** [comment]
I would try different stuff and see what works. I don't know of a better way...

**Comment by sylvain258** [comment]
There is no documentation on expressions except any examples on MapLibre website. 

What would be these different stuffs ?

Thanks

**Reply by HarelM** [reply]
Try modifying the expression until you get something basic working and then modify again.
You can also check the tests in the style-spec repo to see more options, but I'm not sure how helpful are those.
Generally speaking, this is a hard thing to document as there are a lot of options...

**Comment by sylvain258** [comment]
Answer about this question :

I generated symbology with MapBox Studio, in order to have exact syntax.

There may be a bug here.

`'circle-color':
 [
"case",
                [
                  "match",
                  ["get", "annee"],
                  ["NULL"],
                  false,
                  true
                ],
                [
                  "step",
                  ["get", "annee"],
                  "#d1cf8f",
                  1945,
                  "#5f9189",
                  1975,
                  "#ff00ff"
                ],
                "#00A5AD" // The problem is here : Maplibre (Mapbox too) doesn't recognize fallback value (= else)
              ]`
